lib/compress.js

```diff
@@ -3028,63 +3028,49 @@ merge(Compressor.prototype, {
         return this;
     });
 
-    var commutativeOperators = makePredicate("== === != !== * & | ^");
+    var reversibleOperators = makePredicate("+ * & | ^ < > <= >= == != === !==");
```

Comment:
There is extra logic for `+` [down here](https://github.com/mishoo/UglifyJS2/pull/1622/files#diff-230c53e13e3ee367aec7d49470eb42caR3062)